### PR TITLE
docs: convert doc-site theme toggle from nav bar link to segmented control in right sticky page section, persist selection in local storage

### PR DESCRIPTION
**Pull request by danidewitt**
## What did we change?
1. Converted doc-site nav bar theme link to segmented control
2. Moved to right sticky page section
3. Persist theme selection in local storage

## Why are we doing this?
1. It's not a navigational link
2. It's hard to parse if "Use fulfillment theme" means you're currently using marketplace, or you're currently using fulfillment theme
3. Every time you change pages, the theme reverts to the default, this is not obvious and frustrating
4. Designers asked for it (for ease of use for designing with Recipe first)!

## Screenshot(s) / Gif(s):
Screen Shot 2022-03-25 at 5 17 44 PM

Screen Shot 2022-03-25 at 5 17 37 PM

## Checklist

- [ ] Provide test coverage for the changes made
- [x] Create a changeset (`npm run changeset`) with a summary of the changes

[Contributing guidelines](https://ezcater.github.io/recipe/guides/contributing/)

### CODE REVIEW

File: doc-site/src/components/Layout.js [hunk 1]
```diff
@@ -115,8 +113,15 @@ const Layout = ({
       const hasSearchTerms = searchTerms.length;
       const showTableOfContents = hasHeadings && !isComponentGrid && !hasSearchTerms;
 
+      const persistedTheme = localStorage.getItem('recipeTheme');
+      const [activeTheme, setActiveTheme] = useState(persistedTheme || 'fulfillment');
```
Comment (pgrippi) [inline doc-site/src/components/Layout.js:117]: you could `useState(() => localStorage.getItem('recipeTheme') || 'fulfillment')`, which would read from localStorage once on boot rather than every time this component renders, but that's a small optimization!

**Review by pgrippi — APPROVED** [review]

**Review by noranda — APPROVED** [review]
This looks great and is a big improvement! Thanks to @ktlnux for the suggestion!

All tests pass.

**Review by noranda — APPROVED** [review]

**Review by jbelang — APPROVED** [review]

**Review by pgrippi — DISMISSED** [review]

**Review by noranda — APPROVED** [review]

**Review by pgrippi — APPROVED** [review]
